lib/active_record_shards/connection_switcher.rb

```diff
@@ -70,63 +80,78 @@ def on_master_unless(condition, &block)
       on_master_if(!condition, &block)
     end
 
-    def on_master_or_slave(which, &block)
+    def on_master_or_replica(which, &block)
       if block_given?
         on_cx_switch_block(which, &block)
       else
-        MasterSlaveProxy.new(self, which)
+        MasterReplicaProxy.new(self, which)
       end
     end
 
-    # Executes queries using the slave database. Fails over to master if no slave is found.
-    # if you want to execute a block of code on the slave you can go:
-    #   Account.on_slave do
+    def on_master_or_slave(*args, &block)
+      # TODO
+      on_master_or_replica(*args, &block)
+    end
+
+    # Executes queries using the replica database. Fails over to master if no replica is found.
+    # if you want to execute a block of code on the replica you can go:
+    #   Account.on_replica do
     #     Account.first
     #   end
-    # the first account will be found on the slave DB
+    # the first account will be found on the replica DB
     #
     # For one-liners you can simply do
-    #   Account.on_slave.first
+    #   Account.on_replica.first
+    def on_replica(&block)
+      on_master_or_replica(:replica, &block)
+    end
+
     def on_slave(&block)
-      on_master_or_slave(:slave, &block)
+      # TODO
+      on_replica(&block)
     end
 
     def on_master(&block)
-      on_master_or_slave(:master, &block)
+      on_master_or_replica(:master, &block)
     end
 
     # just to ease the transition from replica to active_record_shards
-    alias_method :with_slave, :on_slave
-    alias_method :with_slave_if, :on_slave_if
-    alias_method :with_slave_unless, :on_slave_unless
+    alias_method :with_replica, :on_replica
+    alias_method :with_replica_if, :on_replica_if
+    alias_method :with_replica_unless, :on_replica_unless
```

Comment:
No need to add three new aliases. And we need to keep the old ones. Or delete them: Nobody is still transitioning from the `replica` gem to `active_record_shards`.